src/services/store.service.ts

```diff
@@ -0,0 +1,77 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, firstValueFrom } from 'rxjs';
+import { Item } from '../app/models/item';
+import { CartItem } from 'src/app/models/cartItem';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StoreService {
+  cartItems: CartItem[] = [];
+  totalCost!: number;
+  message$: BehaviorSubject<number> = new BehaviorSubject(0);
+
+  items$ = new BehaviorSubject<Item[]>([]);
+
+  constructor(private readonly http: HttpClient) {}
+  // old method with promise
+  // async getAllItems(): Promise<Item[]> {
+  //   const items = await firstValueFrom(
+  //     this.http.get<Item[]>(`${environment.apiUrl}`)
+  //   );
+  //   return items;
+  // }
+
+  fetchAll() {
+    this.http
```

Comment:
we should be storing these subscriptions so we can unsubscribe from them. Otherwise, we will cause a memory leak.